nameable.rb

```diff
@@ -0,0 +1,30 @@
+class Nameable
+  def correct_name(name)
+    raise NotImplementedError
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0, 10]
+  end
+end
```

Comment:
- [ ] [OPTIONAL] Kindly use separate files for `BaseDecorator`, `CapitalizeDecorator`, and `TrimmerDecorator` and use the `require` or `require_relative` instead of plotting the `nameable` file with all classes.